src/source/ObjcMarshal.scala

```diff
@@ -66,7 +66,7 @@ class ObjcMarshal(spec: Spec) extends Marshal(spec) {
         }
       case DRecord =>
         val r = d.body.asInstanceOf[Record]
-        val prefix = if (r.ext.objc) "../" else ""
+        val prefix = if (r.ext.objc) spec.objcExtendedRecordIncludePrefix else ""
         List(ImportRef(q(spec.objcIncludePrefix + prefix + headerName(d.name))))
```

Comment:
Looks like this line will be combining the new prefix with the old objcIncludePrefix.  I think you want to make it use only one or the other, as you did in ObjcGenerator.scala.